CMakeLists.txt

```diff
@@ -248,24 +249,8 @@ if (COMPILER_SUPPORTS_AVX2)
   add_compile_options(${COMPILER_AVX2_FLAG})
 endif()
 
-# HACK: Set the sanitizer configuration globally after the
-# superbuild has finished. We want to enable sanitization for
-# TileDB, but not our dependencies.
-#
-# The reasoning for setting this globally is due to false positives
-# in the `AddressSanitizerContainerOverflow` checks [1].
-#
-# [1] https://github.com/google/sanitizers/wiki/AddressSanitizerContainerOverflow#false-positives
-if (SANITIZER)
-  string(TOLOWER ${SANITIZER} SANITIZER)
-  if (NOT SANITIZER MATCHES "^(address|memory|leak|thread|undefined)$")
-    message(FATAL_ERROR "Unknown clang sanitizer: ${SANITIZER})")
-  else()
-    message(STATUS "The TileDB library is compiled with sanitizer ${SANITIZER} enabled")
-  endif()
-  add_compile_options(-DTILEDB_SANITIZER=${SANITIZER})
-  add_compile_options(-g -fno-omit-frame-pointer -fno-optimize-sibling-calls -fsanitize=${SANITIZER})
-  add_link_options(-fsanitize=${SANITIZER})
+if(SANITIZER)
```

Comment:
Did you run a manual test to see if this code ever runs? It looks like this variable is never set anywhere, which means none of the code guarded by it will ever run.